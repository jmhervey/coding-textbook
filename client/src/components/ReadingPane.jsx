import React, {useState, useEffect} from 'react';

const ReadingPane = React.memo( props => {

  const [loading, setLoading] = useState(true)
  const [html, setHtml] = useState("")
  /*
   * Makes API call that embeds hypothes into source of the page, allowing notes on the page
   */
  useEffect( () => {
    fetch('/api/reading', {
      method: 'POST',
      headers: {'Content-Type': 'application/json'},
      body: JSON.stringify({url: props.url})
    })
    .then(response => {
      if(response.ok) return response.text()
      throw response
    })
    .then(html => {
      const newHtml = html.replace('</body>', `<p id="reading-bottom" data-id="${props.id}"></p><script type="text/javascript" src="http://localhost:3000/api/embed.js"></script></body>`)
      setHtml(html);
      setLoading(false);
    })
  })

  if(loading) return <>Loading...</>

  return (
    <iframe id="reading-pane" title="The Assigned Reading" src={props.url} srcDoc={html} width="100%" height="100%" />
  )

})




export default ReadingPane;