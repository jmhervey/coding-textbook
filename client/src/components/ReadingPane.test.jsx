import React from 'react';
import { render } from '@testing-library/react';
import ReadingPane from './ReadingPane';
import fetchMock from "jest-fetch-mock";

// Configure fetch mock
fetchMock.enableMocks();

describe('ReadingPane', () => {
  test('Unit Test for the ReadingPane component', () => {

    const items = [
      { id: 1, title: 'Article 1', content: 'Content 1' },
      { id: 2, title: 'Article 2', content: 'Content 2' },
    ];

    // Render the ReadingPane component
    const { container } = render(<ReadingPane items={items} />);

    // Test if the rendered component matches the saved snapshot
    expect(container.firstChild).toMatchSnapshot();
  });
});
