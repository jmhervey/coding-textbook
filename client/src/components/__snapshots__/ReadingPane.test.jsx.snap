// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReadingPane renders the ReadingPane component 1`] = `Loading...`;
